### PR TITLE
T18078 test group clean up

### PR DESCRIPTION
**Pull request by gctucker**
Removed unused fields in TestGroupDocument and clean up some related source code.

### CODE REVIEW

**Comment by gctucker** [comment]
Tested OK on staging, got `baseline` tests report and a bisection result for `baseline.login`.

```
kernelci/staging.kernelci.org baseline: 51 runs, 5 regressions (staging-20200110.0)

Test results summary
--------------------

run | platform                     | arch   | lab          | compiler | defconfig                    | results
----+------------------------------+--------+--------------+----------+------------------------------+--------
1   | am335x-boneblack             | arm    | lab-baylibre | clang-8  | multi_v7_defconfig           | 0/1    
2   | am335x-boneblack             | arm    | lab-baylibre | gcc-8    | multi_v7_defconfig           | 0/1    
3   | hifive-unleashed-a00         | riscv  | lab-baylibre | gcc-8    | defconfig                    | 1/1    
4   | imx6q-sabrelite              | arm    | lab-baylibre | clang-8  | multi_v7_defconfig           | 5/5    
5   | imx6q-sabrelite              | arm    | lab-baylibre | gcc-8    | multi_v7_defconfig           | 5/5    
6   | imx8mn-ddr4-evk              | arm64  | lab-baylibre | gcc-8    | defconfig                    | 5/5    
7   | meson-g12a-sei510            | arm64  | lab-baylibre | gcc-8    | defconfig                    | 5/5    
8   | meson-g12a-x96-max           | arm64  | lab-baylibre | gcc-8    | defconfig                    | 5/5    
9   | meson-g12b-odroid-n2         | arm64  | lab-baylibre | gcc-8    | defconfig                    | 5/5    
10  | meson-gxbb-p200              | arm64  | lab-baylibre | gcc-8    | defconfig                    | 5/5    
11  | meson-gxbb-p200              | arm64  | lab-baylibre | gcc-8    | defconfig+CON...BIG_ENDIAN=y | 5/5    
12  | meson-gxl-s805x-libretech-ac | arm64  | lab-baylibre | gcc-8    | defconfig                    | 5/5    
13  | meson-gxl-s805x-libretech-ac | arm64  | lab-baylibre | gcc-8    | defconfig+CON...BIG_ENDIAN=y | 5/5    
14  | meson-gxl-s805x-p241         | arm64  | lab-baylibre | gcc-8    | defconfig                    | 5/5    
15  | meson-gxl-s905d-p230         | arm64  | lab-baylibre | gcc-8    | defconfig                    | 5/5    
16  | meson-gxl-s905x-khadas-vim   | arm64  | lab-baylibre | gcc-8    | defconfig+CON...BIG_ENDIAN=y | 5/5    
17  | meson-gxl-s905x-libretech-cc | arm64  | lab-baylibre | gcc-8    | defconfig                    | 5/5    
18  | meson-gxl-s905x-libretech-cc | arm64  | lab-baylibre | gcc-8    | defconfig+CON...BIG_ENDIAN=y | 5/5    
19  | meson-gxm-khadas-vim2        | arm64  | lab-baylibre | gcc-8    | defconfig                    | 5/5    
20  | meson-gxm-khadas-vim2        | arm64  | lab-baylibre | gcc-8    | defconfig+CON...BIG_ENDIAN=y | 5/5    
21  | meson-gxm-q200               | arm64  | lab-baylibre | gcc-8    | defconfig                    | 5/5    
22  | meson-gxm-q200               | arm64  | lab-baylibre | gcc-8    | defconfig+CON...BIG_ENDIAN=y | 0/1    
23  | meson-sm1-sei610             | arm64  | lab-baylibre | gcc-8    | defconfig                    | 5/5    
24  | meson8b-odroidc1             | arm    | lab-baylibre | clang-8  | multi_v7_defconfig           | 0/1    
25  | meson8b-odroidc1             | arm    | lab-baylibre | gcc-8    | multi_v7_defconfig           | 0/1    
26  | omap4-panda                  | arm    | lab-baylibre | clang-8  | multi_v7_defconfig           | 0/1    
27  | omap4-panda                  | arm    | lab-baylibre | gcc-8    | multi_v7_defconfig           | 0/1    
28  | qemu_arm-virt-gicv2          | arm    | lab-baylibre | clang-8  | multi_v7_defconfig           | 1/1    
29  | qemu_arm-virt-gicv2          | arm    | lab-baylibre | gcc-8    | multi_v7_defconfig           | 1/1    
30  | qemu_arm-virt-gicv3          | arm    | lab-baylibre | clang-8  | multi_v7_defconfig           | 1/1    
31  | qemu_arm-virt-gicv3          | arm    | lab-baylibre | gcc-8    | multi_v7_defconfig           | 1/1    
32  | qemu_arm64-virt-gicv2        | arm64  | lab-baylibre | gcc-8    | defconfig                    | 5/5    
33  | qemu_arm64-virt-gicv3        | arm64  | lab-baylibre | gcc-8    | defconfig                    | 5/5    
34  | qemu_i386                    | i386   | lab-baylibre | gcc-8    | i386_defconfig               | 5/5    
35  | qemu_x86_64                  | x86_64 | lab-baylibre | gcc-8    | x86_64_defconfig             | 5/5    
36  | sun4i-a10-olinuxino-lime     | arm    | lab-baylibre | clang-8  | multi_v7_defconfig           | 5/5    
37  | sun4i-a10-olinuxino-lime     | arm    | lab-baylibre | gcc-8    | multi_v7_defconfig           | 5/5    
38  | sun50i-a64-pine64-plus       | arm64  | lab-baylibre | gcc-8    | defconfig+CON...BIG_ENDIAN=y | 5/5    
39  | sun50i-h5-lib...ch-all-h3-cc | arm64  | lab-baylibre | gcc-8    | defconfig                    | 5/5    
40  | sun5i-a13-olinuxino-micro    | arm    | lab-baylibre | clang-8  | multi_v7_defconfig           | 5/5    
41  | sun5i-a13-olinuxino-micro    | arm    | lab-baylibre | gcc-8    | multi_v7_defconfig           | 5/5    
42  | sun7i-a20-olinuxino-lime2    | arm    | lab-baylibre | clang-8  | multi_v7_defconfig           | 5/5    
43  | sun7i-a20-olinuxino-lime2    | arm    | lab-baylibre | gcc-8    | multi_v7_defconfig           | 5/5    
44  | sun8i-h2-plus...ch-all-h3-cc | arm    | lab-baylibre | clang-8  | multi_v7_defconfig           | 5/5    
45  | sun8i-h2-plus...ch-all-h3-cc | arm    | lab-baylibre | gcc-8    | multi_v7_defconfig           | 5/5    
46  | sun8i-h2-plus-orangepi-r1    | arm    | lab-baylibre | clang-8  | multi_v7_defconfig           | 5/5    
47  | sun8i-h2-plus-orangepi-r1    | arm    | lab-baylibre | gcc-8    | multi_v7_defconfig           | 5/5    
48  | sun8i-h2-plus-orangepi-zero  | arm    | lab-baylibre | clang-8  | multi_v7_defconfig           | 5/5    
49  | sun8i-h2-plus-orangepi-zero  | arm    | lab-baylibre | gcc-8    | multi_v7_defconfig           | 5/5    
50  | sun8i-h3-libretech-all-h3-cc | arm    | lab-baylibre | clang-8  | multi_v7_defconfig           | 5/5    
51  | sun8i-h3-libretech-all-h3-cc | arm    | lab-baylibre | gcc-8    | multi_v7_defconfig           | 5/5    

[...]
```

File: app/models/tests/test_test_group_model.py [hunk 1]
```diff
@@ -50,13 +50,10 @@ def test_group_doc_to_dict(self):
         test_group.cross_compile = "cross-compile"
```
Comment (gctucker) [inline app/models/tests/test_test_group_model.py]: @khilman The fields being removed below don't appear to have ever been used anywhere in production.  Was it used on an other instance that you know of?

File: app/utils/kci_test/__init__.py [hunk 2]
```diff
@@ -446,82 +433,80 @@ def _update_test_group_doc_ids(group_doc, database):
...
 
     if build_doc:
         doc_get = build_doc.get
-        group_doc.build_id = doc_get(models.ID_KEY, None)
+        group_doc.build_id = doc_get(models.ID_KEY)
 
         # In case we do not have the job_id key with the previous search.
-        if all([not group_doc.job_id, doc_get(models.JOB_ID_KEY, None)]):
-            group_doc.job_id = doc_get(models.JOB_ID_KEY, None)
+        if all([not group_doc.job_id, doc_get(models.JOB_ID_KEY)]):
```
Comment (mgalka) [inline app/utils/kci_test/__init__.py]: Wouldn't that be easier to read and understand with a simple condition like 
`if not group_doc.job_id and doc_get(models.JOB_ID_KEY, None):`?

Comment (gctucker) [reply app/utils/kci_test/__init__.py]: Yes, that seems better.  I didn't try to simplify the code other than remove unnecessary `None` default values and unused fields in the test group model.

Comment (gctucker) [reply app/utils/kci_test/__init__.py]: @mgalka I've actually simplified this quite a bit, see new version.

Comment (mgalka) [reply app/utils/kci_test/__init__.py]: This is even better :) Thanks

**Pull request merged**